sympy/functions/elementary/piecewise.py

```diff
@@ -1045,6 +1065,79 @@ def rewrite(cond):
 
             return result
 
+    def _eval_rewrite_as_sign(self, *args, factor=True, **kwargs):
+        """
+        factor : boolean
+            If True, try a pre-processing factorization to
+            increase the chances of success of the transformation.
+        """
+
+        from sympy.sets.sets import Interval, FiniteSet
+        from sympy import oo
+        from sympy.functions.elementary.complexes import sign
+
+        # preprocess
+        fact = 1
+        if factor:
+            fact_expr = Piecewise(*args).factor()
+            if fact_expr.is_Mul:
+                fact_args = fact_expr.args
+                piecewises = [f for f in fact_args if f.is_Piecewise]
+                if len(piecewises) != 1:
+                    return
+                from sympy.core.mul import Mul
+                args = piecewises.pop().args
+                fact = Mul(*[f for f in fact_args if not f.is_Piecewise])
+        args = list(args)
+
+        # the comparison elements are reported to 0
+        for i, (expr, cond) in list(enumerate(args)):
+            if isinstance(cond, (GreaterThan, LessThan,
```

Comment:
`Unequality`?

Reply:
Just to clarify (maybe I just do not understand the code...), but Unequality is ~=, which wondered if it should be there.

If you want to test for all (==, <, <=, >, >=, ~=) you can use `Relational`.

Reply:
The aim here is just to reduce it to a comparison with 0 in order to facilitate the change of variable. Maybe there are even other classes besides Relational? I'm not very familiar with sympy. I'm interested in all these little tips! thank you ;)